### PR TITLE
rockchip: add NanoPi R4S Enterprise Edition build

### PR DESCRIPTION
**Pull request by 1715173329**
FriendlyElec renamed the NanoPi R4S board with EEPROM (mac address) to "enterprise" edition, and it was added as a "new" board in upstream kernel.

This patch switched to use that upstreamed dts and removed local EEPROM patch.

### CODE REVIEW

File: target/linux/rockchip/armv8/base-files/etc/board.d/02_network [hunk 1]
```diff
@@ -60,18 +40,19 @@ rockchip_setup_macs()
...
+		;;
+	friendlyarm,nanopi-r4s-enterprise)
+		wan_mac=$(get_mac_binary "/sys/bus/i2c/devices/2-0051/eeprom" 0xfa)
+		lan_mac=$(macaddr_setbit_la "$wan_mac")
 		;;
 	friendlyarm,nanopi-r5c)
 		wan_mac=$(macaddr_generate_from_mmc_cid mmcblk*)
 		lan_mac=$(macaddr_add "$wan_mac" 1)
 		;;
-	friendlyarm,nanopi-r5s)
```
Comment (RaylynnKnight) [inline target/linux/rockchip/armv8/base-files/etc/board.d/02_network:71]: Why is the entry for nanopi-r5s removed here?  Your comment makes no mention of this device.

Comment (1715173329) [reply target/linux/rockchip/armv8/base-files/etc/board.d/02_network:71]: https://github.com/openwrt/openwrt/pull/14657/commits/f9c254c33a723f810f18bb8d0bd770a639a820cf#diff-1ce315e56a6173a0798c249e226c43413ee01c9b5e941f56886d938001d3228fR44

**Comment by PolynomialDivision** [comment]
So there are now two versions? One with eeprom and one without? Can you resolve the conflict? Thanks! :)

**Comment by 1715173329** [comment]
> So there are now two versions? One with eeprom and one without?

Yes.

> Can you resolve the conflict?

Done! ;)

**Comment by PolynomialDivision** [comment]
Thanks! I will wait for CI again but looks good. ^^

**Pull request merged**